wallet/src/account/operations/transaction/prepare_output.rs

```diff
@@ -167,9 +160,19 @@ impl AccountHandle {
         // We might have added more unlock conditions, so we check the minimum storage deposit again and update the
         // amounts if needed
         if second_output.amount() < required_storage_deposit {
-            third_output_builder = third_output_builder.with_amount(required_storage_deposit)?;
+            let mut new_sdr_amount = required_storage_deposit - options.amount;
+            let minimum_storage_deposit = minimum_storage_deposit_basic_output(&rent_structure, &None, token_supply)?;
+            let mut final_output_amount = required_storage_deposit;
+            if required_storage_deposit - options.amount < minimum_storage_deposit {
```

Comment:
Maybe do the opposite check to avoid the underflow ?

Reply:
Changed to prevent errors in the future if we want to change something, but it's safe because of `if second_output.amount() < required_storage_deposit {`